core/src/main/java/org/pentaho/platform/scheduler2/action/SchedulerOutputPathResolver.java

```
@@ -110,23 +111,36 @@ public void setActionUser( String actionUser ) {
   @Override
   public String resolveOutputFilePath() throws SchedulerException {
 
-    // IMPROVEMENT: This does not belong here. Move to the caller.
-    boolean scheduleAllowed = isScheduleAllowed();
-    if ( !scheduleAllowed ) {
-      throw new SchedulerException( Messages.getInstance().getString(
-        "QuartzScheduler.ERROR_0009_SCHEDULING_IS_NOT_ALLOWED_AFTER_CHANGE", getJobName(), getActionUser() ) );
-    }
+    try {
+      // IMPROVEMENT: This does not belong here. Move to the caller.
+      boolean scheduleAllowed = isScheduleAllowed();
+      if (!scheduleAllowed) {
+        throw new SchedulerException(Messages.getInstance().getString(
+                "QuartzScheduler.ERROR_0009_SCHEDULING_IS_NOT_ALLOWED_AFTER_CHANGE", getJobName(), getActionUser()));
+      }
 
-    // Enclose validation logic in the context of the job creator's session, not the current session
-    return runAsUser( this::resolveOutputFilePathCore );
+      // Enclose validation logic in the context of the job creator's session, not the current session
+      return runAsUser( this::resolveOutputFilePathCore );
+    } catch ( SchedulerException schedulerException ) {
+      throw new SchedulerException( schedulerException );
```

I'm not sure what you are trying to do here.

You almost never wan to do swallow all exceptions (unless with good reason):

```
} catch ( Exception e ) {
      logger.error( e.getMessage(), e );
    }
```

Are you trying to differentiate between normal SchedulerException and one caused by the fall back logic?

---

I was trying to throw a scheduler exception when fallback logic doesn't permit schedule to fallback and leaving other exceptions for a chance to other code to execute .
 https://github.com/pentaho/pentaho-scheduler-plugin/blob/ff2df57a9cb74c7b552ae8d708231a449f244e8b/core/src/main/java/org/pentaho/platform/scheduler2/action/ActionRunner.java#L158

Will be creating a new FallBackSchedulerException as mentioned https://github.com/pentaho/pentaho-scheduler-plugin/pull/277#discussion_r2001465053

---

Addressed. Create a custom Exception class FallBackSchedulerException .